statsmodels/genmod/families/family.py

```diff
@@ -1432,3 +1432,193 @@ def resid_anscombe(self, endog, mu):
+    Still under development.
+    """
+    links = [L.log, L.Power]
+    # TODO: add the ability to use the power links with an if test
+    # similar to below
+    variance = V.Power
+    safe_links = [L.log, L.Power]
+
+    def __init__(self, link=None, var_power=1., link_power=0):
+        self.var_power = var_power
+        self.link_power = link_power
+        self.variance = V.Power(power=var_power * 1.)
+        if link_power != 0 and not ((link is L.Power) or (link is None)):
+            msg = 'link_power of {} not supported specified link'
+            msg = msg.format(link_power)
+            raise ValueError(msg)
+        if (link_power == 0) and ((link is None) or (link is L.Log)):
+            self.link = L.log()
+        elif link_power != 0:
+            self.link = L.Power(power=link_power * 1.)
+        else:
+            self.link = link()
+
+    def _clean(self, x):
+        """
+        Helper function to trim the data so that is in (0,inf)
+
+        Notes
+        -----
+        The need for this function was discovered through usage and its
+        possible that other families might need a check for validity of the
+        domain.
+        """
+        return np.clip(x, 0, np.inf)
+
+    def deviance(self, endog, mu, freq_weights=1., scale=1.):
+        r"""
+        Returns the value of the deviance function.
+
+        Parameters
+        -----------
+        endog : array-like
+            Endogenous response variable
+        mu : array-like
+            Fitted mean response variable
+        freq_weights : array-like
+            1d array of frequency weights. The default is 1.
+        scale : float, optional
+            An optional scale argument. The default is 1.
+
+        Returns
+        -------
+        deviance : float
+            Deviance function as defined below
+
+        Notes
+        -----
+        Not yet...
+        """
+        endog1 = np.clip(endog, 0.1, np.inf)
+        p = self.var_power
+        if p == 1:
+            dev = mu - endog * (1 + np.log(mu / endog1))
+        elif p == 2:
+            dev = -1 + (endog / mu) + np.log(mu / endog1)
+        else:
+            dev = (endog1 ** (2 - p) / ((1 - p) * (2 - p)) -
+                   endog1 * mu ** (1-p) / (1 - p) + mu ** (2 - p) / (2 - p))
+        return np.sum(2 * freq_weights * dev)
+
+    def resid_dev(self, endog, mu, scale=1.):
+        r"""
+        Negative Binomial Deviance Residual
+
+        Parameters
+        ----------
+        endog : array-like
+            `endog` is the response variable
+        mu : array-like
+            `mu` is the fitted value of the model
+        scale : float, optional
+            An optional argument to divide the residuals by scale. The default
+            is 1.
+
+        Returns
+        --------
+        resid_dev : array
+            The array of deviance residuals
+
+        Notes
+        -----
+        Not yet...
+        """
+        endog1 = np.clip(endog, 0.1, np.inf)
```

Comment:
I rethought this... there's some new logic. I still would like to have this to idiot proof cases where they select `p>=2`, but have at least 1 `endog` of 0... I'm using `FLOAT_EPS` instead.